File: src/main/java/io/github/tanguygab/advancedbungeeexpansion/ServerInfo.java
package io.github.tanguygab.advancedbungeeexpansion;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@AllArgsConstructor
public class ServerInfo {

    @Getter private final String name;
    @Getter @Setter private boolean status;
    @Getter @Setter private String motd;
    @Getter @Setter private List<String> players;

    public ServerInfo(String name, List<String> players) {
        this(name,false,"No MOTD",players);
    }

    public int getPlayerCount() {
        return players.size();
    }
}

File: src/main/java/io/github/tanguygab/advancedbungeeexpansion/ServerInfo.kt
package io.github.tanguygab.advancedbungeeexpansion

import lombok.AllArgsConstructor
import lombok.Getter
import lombok.Setter

@Getter
@AllArgsConstructor
class ServerInfo {
    var name: String = ""
    var status = false
    var motd: String? = null
    var players: List<String>? = null

    constructor(name: String, status: Boolean, motd: String, players: List<String>) {
        this.name = name
        this.status = status
        this.motd = motd
        this.players = players
    }

    constructor(name: String, players: List<String>) {
        this.name = name ?: ""
        this.status = false
        this.motd = "No MOTD"
        this.players = players

    }

    val playerCount: Int
        get() = players!!.size
}
